java_grpc_library.bzl

```diff
@@ -62,7 +62,20 @@ java_rpc_toolchain = rule(
 def _path_ignoring_repository(f):
     if len(f.owner.workspace_root) == 0:
```

Comment:
Hmm... you're probably right that something is broken there, as it had only been tested when a single workspace was in place.

Am I right in thinking that this PR doesn't break anything? It mainly doesn't work in a case? If so, I think I may merge this and then do a follow-up (I'm not 100% confident I understand the change you're suggesting, so it may also be a good change for you to do). This version of the patch started in https://github.com/bazelbuild/bazel/pull/8814 and has gone in internally and in https://github.com/bazelbuild/bazel-buildfarm/pull/272 . Those other places are probably fine with the limitation, but it'd be a bit clearer to track as a separate commit.

Reply:
The change shouldn't break anything and the problem is fixable.
Is it actually used somewhere with the `*import_prefix` arg set?

Reply:
It doesn't break anything that I know of (that is to say, anything in the Bazel presubmit :) )

That said, you have a point -- however, since this change is already submitted within Google, how about merging this and creating a followup change with the suggested grammar fix and moving this check after the `_virtual_imports` one?